nestkernel/nest_types.h

```diff
@@ -59,6 +59,50 @@ namespace nest
  * different architectures (e.g. 32 or 64 bit).
  */
 
+// constexpr-functions for convenient compile-time generation of the bit-masks
+// and bit-constants. An ill-defined length or size will cause a compile-time
+// error, e.g., num_bits to be shifted exceeds the sizeof(<datatype>) * 8.
+constexpr uint64_t
+generate_bit_mask( const uint8_t num_bits, const uint8_t bit_position )
+{
+  return (
+    ( ( static_cast< uint64_t >( 1 ) << num_bits ) - 1 ) << bit_position );
+}
+
+constexpr uint64_t
+generate_max_value( const uint8_t num_bits )
+{
+  return ( ( static_cast< uint64_t >( 1 ) << num_bits ) - 1 );
+}
+
+/*
+ * Sizes of bitfields used in various classes in the kernel.
+ */
+#if TARGET_BITS_SPLIT == TARGET_BITS_SPLIT_STANDARD
+constexpr uint8_t NUM_BITS_RANK = 18U;
+constexpr uint8_t NUM_BITS_TID = 9U;
+constexpr uint8_t NUM_BITS_SYN_ID = 9U;
+#elif TARGET_BITS_SPLIT == TARGET_BITS_SPLIT_HPC
+constexpr uint8_t NUM_BITS_RANK = 20U;
+constexpr uint8_t NUM_BITS_TID = 10U;
+constexpr uint8_t NUM_BITS_SYN_ID = 6U;
+#endif
+constexpr uint8_t NUM_BITS_LCID = 27U;
+constexpr uint8_t NUM_BITS_PROCESSED_FLAG = 1U;
+constexpr uint8_t NUM_BITS_MARKER_SPIKE_DATA = 2U;
+constexpr uint8_t NUM_BITS_LAG = 14U;
+constexpr uint8_t NUM_BITS_DELAY = 21U;
+constexpr uint8_t NUM_BITS_GID = 62U;
+
+/*
+ * Maximally allowed values for bitfields
+ */
+constexpr uint64_t MAX_LCID = generate_max_value( NUM_BITS_LCID );
+constexpr int MAX_RANK = generate_max_value( NUM_BITS_RANK );
```

Comment:
```suggestion
constexpr uint64_t MAX_RANK = generate_max_value( NUM_BITS_RANK );
```

Reply:
@jakobj The constexpr function now returns an uint64_t. Since MAX_nnn is only used in the assertions uint64_t is the better choice.